### PR TITLE
Front/translate refactor

### PR DESCRIPTION
**Pull request by Arthi-chaud**
Add a "Translate" React component which would update the text on language change.

### CODE REVIEW

File: front/components/forms/signupform.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // a form for sign up
 
 import React from "react";
-import { translate } from "../../i18n/i18n";
+import { Translate, translate } from "../../i18n/i18n";
```
Comment (Octopus773) [inline front/components/forms/signupform.tsx:4]: Does the reload of `Translate` composant reload the `translate` function calls (like the function in this file are going to work as intended ? )

File: front/components/SongCardGrid.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { Heading, VStack } from 'native-base';
 type SongCardGrid = {
 	songs: Parameters<typeof SongCard>[0][];
 	maxItemPerRow?: number,
-	heading?: string
+	heading?: JSX.Element,
```
Comment (Octopus773) [inline front/components/SongCardGrid.tsx:9]: Why this change here ? Because the Translate JSX is supposed to give us a string

Comment (AmauryDanisCousandier) [reply front/components/SongCardGrid.tsx:9]: je me pose la même question du coup

File: front/components/forms/signupform.tsx (hunk 1 above)
Comment (Arthi-chaud) [reply front/components/forms/signupform.tsx:4]: The "Translate" component relies on a hook to retrieve the current language. Each time this changes, the component will be re-rendered, ie will recall the 'translate' function

File: front/components/SongCardGrid.tsx (hunk 2 above)
Comment (Octopus773) [reply front/components/SongCardGrid.tsx:9]: OK for the reusability of the component, but for the text update the component would have been fine because the text parameter would have to change and thus re-render the component

**Review by Octopus773 — APPROVED** [review]
